crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -98,22 +96,30 @@ pub(crate) fn use_def_map<'db>(db: &'db dyn Db, scope: ScopeId<'db>) -> Arc<UseD
 }
 
 /// Returns all attribute assignments for a specific class body scope.
-///
-/// Using [`attribute_assignments`] over [`semantic_index`] has the advantage that
-/// Salsa can avoid invalidating dependent queries if this scope's instance attributes
-/// are unchanged.
-#[salsa::tracked]
-pub(crate) fn attribute_assignments<'db>(
+pub(crate) fn attribute_assignments<'db, 's>(
     db: &'db dyn Db,
     class_body_scope: ScopeId<'db>,
-) -> Option<Arc<AttributeAssignments<'db>>> {
+    name: &'s str,
+) -> impl Iterator<Item = BindingWithConstraints<'db, 'db>> + use<'s, 'db> {
     let file = class_body_scope.file(db);
     let index = semantic_index(db, file);
-
-    index
-        .attribute_assignments
-        .get(&class_body_scope.file_scope_id(db))
-        .cloned()
+    let class_scope_id = class_body_scope.file_scope_id(db);
+
+    DescendantsIter::new(index, class_scope_id)
+        .filter_map(move |(file_scope_id, maybe_method)| {
+            if maybe_method
+                .parent()
+                .is_none_or(|parent| parent != class_scope_id)
+            {
+                return None;
+            }
```

Comment:
This logic is already implemented as `ChildrenIter`, should we just use that instead of `DescendantsIter`?

(There's no efficiency improvement, the implementation of `ChildrenIter` is effectively the same, it just avoids re-implementing the logic.)